test/conformance/api/volumes_test.go

```diff
@@ -72,18 +73,97 @@ func TestConfigMapVolume(t *testing.T) {
 			LocalObjectReference: corev1.LocalObjectReference{
 				Name: configMap.Name,
 			},
+			Optional: ptr.Bool(false),
 		},
-	},
-	)
+	})
 
-	// Setup initial Service
-	if _, err := test.CreateRunLatestServiceReady(t, clients, &names, &test.Options{}, withVolume); err != nil {
+	withBadVolume := WithVolume("blah", "/does/not/matter", corev1.VolumeSource{
+		ConfigMap: &corev1.ConfigMapVolumeSource{
+			LocalObjectReference: corev1.LocalObjectReference{
+				Name: "does-not-exist",
+			},
+			Optional: ptr.Bool(true),
+		},
+	})
 
+	// Setup initial Service
+	if _, err := test.CreateRunLatestServiceReady(t, clients, &names, &test.Options{}, withVolume, withBadVolume); err != nil {
 		t.Fatalf("Failed to create initial Service %v: %v", names.Service, err)
 	}
 
 	// Validate State after Creation
+	if err = validateRunLatestControlPlane(t, clients, names, "1"); err != nil {
+		t.Error(err)
+	}
+
+	if err = validateRunLatestDataPlane(t, clients, names, text); err != nil {
+		t.Error(err)
+	}
+}
+
+// TestProjectedConfigMapVolume tests that we echo back the appropriate text from the ConfigMap volume.
+func TestProjectedConfigMapVolume(t *testing.T) {
+	t.Parallel()
+	clients := test.Setup(t)
+
+	names := test.ResourceNames{
+		Service: test.ObjectNameForTest(t),
+		Image:   "hellovolume",
+	}
+
+	text := test.AppendRandomString("hello-volumes-")
+
+	// Create the ConfigMap with random text.
+	configMap, err := clients.KubeClient.Kube.CoreV1().ConfigMaps(test.ServingNamespace).Create(&corev1.ConfigMap{
```

Comment:
Can you follow the pattern here ( https://github.com/knative/serving/commit/c26d86a69f5b77c6acd1c570047e4084e5c5a7af#diff-7adb269f0ed4dedc3fa8e1157f25dc98 ) to move ConfigMap and Secret creation outside of the test itself?

If you think it is more appropriate to do all of them in this file in a single follow-up I am okay with that as well.

Reply:
I think we should do the whole file at once.